examples/inference/README.md

```diff
@@ -0,0 +1,175 @@
+* PiPPY supports both manual splitting and auto split.
+* Auto split uses `split_policy` and support both `equal_size` and `threshod` policies, the name are self-explanatory.
+* PiPPY use FX to trace and split the model.
+
+## Setting you need to care about
+
+* pp_group_size configure the number of pipeline parallelism group, meaning essentially on how many gpus our model to need be splitted and form a pipeline.
+
+**Main difference between Pippy for training and inference is we dont need to call the init_data_parallel API in the inference. The reason is DDP init is only needed if we need backward pass which is not the case for inference.**
+
+
+For example to serve two copies of the model with 8 gpus, assuming we can serve a full copy of a model splitted on 4 gpus, we set the pp_group_size=4. In this case the pipeline driver is running from rank 0 for the first copy and rank4 for the second copy of the model. In the run_master funtion after initializing the stage after splitting `model_pipe.defer_stage_init(args.device)' need to check if the rank=!0 or 4 to return. As shown below:
+```
+model_pipe.defer_stage_init(args.device)
+torch.distributed.barrier(args.pp_group)
+if args.rank!=0 or args.rank!=4:
+        return 
+```
+        
+Then the resulted models (2 copies) will look like:
+
+<img src="https://user-images.githubusercontent.com/9162336/206811740-1d5d4d7a-7018-4e9b-8ab7-cad030e2981a.png" alt="drawing" width="400"/>
+
+
+
+
+## How to Use PiPPY for inference
+
+**Define a function such as run_master() and add the followings to it.**
+
+We use a HuggingFace T5 model as the running example here. The `HF_inference.py` also support HF OPT, Bloom, RegNet models as well. Make sure to specifiy the model name as follows ` python HF_inference.py --model_name "facebook/opt-2.7b" `
+
+* Load your model normally on CPU
+
+example:
+
+` t5 = AutoModelForSeq2SeqLM.from_pretrained('t5-11b', use_cache=False) `
+
+* The "MULTI_USE_PARAM_CONFIG" addresses corner cases where if an Operation would need a parameter that has been placed on a different GPU. In this case with setting we can either REPLICATE or TRANSMIT that paramter to the GPU/rank that needs it. 
+
+ `MULTI_USE_PARAM_CONFIG = MultiUseParameterConfig.REPLICATE if args.replicate else MultiUseParameterConfig.TRANSMIT`
+
+*  Setup the model split policy
+
+```
+from pippy import split_on_size_threshold, split_into_equal_size
+
+if args.auto_split == "threshold":
+        split_policy = split_on_size_threshold(490 * 1e6)
+elif args.auto_split == "equal_size":
+        split_policy = split_into_equal_size(number_of_workers)
+```
+* Make the concerete args, this is required for FX tracing by giving it concerete inputs, this will be used to run the input through the model to check if there any control flow. For now control flow is not supported in FX tracing, we are working on integrating Torch Dynamo to make this more flexible. 
```

Comment:
Maybe add an "(Optional)" marker for this comment re concerete args?

Reply:
"is required for FX tracing" is not entirely true.
Maybe we can change it to:
"If the model has inside an if-else condition, the concrete args can help FX determine which path to trace".